### PR TITLE
Cache S3Directory listAll result

### PR DESCRIPTION
**Pull request by adamyy**
* Assume static indexes, cache S3Directory `listAll` result
* Use RequestHandler interface to handle boilerplate
* Bump Java source/build version to 11 to match Lambda deployment version
* Parameterized DynamoDB table name

### CODE REVIEW

**Comment by shaneding** [comment]
Sorry haven't used S3 too much, but what does caching the S3Directory do, does it just allow faster lambda response time on future instances? Otherwise lgtm

**Comment by adamyy** [comment]
@shaneding If you assume your indexes are static (i.e., read-only) and never change, then an `ls` will always return the same result. Therefore the result of `listAll` can be cached to save the amount of round trips between S3 and Lambda.

File: src/main/java/io/anlessini/SearchLambda.java [hunk 1]
```diff
@@ -27,72 +31,68 @@
...
+      }
+    }
 
-    String hits = search(qstring, context);
-    outputStream.write(("{\"statusCode\": 200, \"headers\": { \"Access-Control-Allow-Origin\": \"*\" }," +
-            "\"body\": " + mapper.writeValueAsString(hits) + "}").getBytes());
+    response.setHeaders(Map.of(
+        "Content-Type", "application/json"
+    ));
+    return response;
```
Comment (shaneding) [inline src/main/java/io/anlessini/SearchLambda.java]: Does the structure of the json that is being returned change at all? It's fine if it does, but some stuff might have to be changed on the anlessini-demo repo if it's different.

Comment (adamyy) [reply src/main/java/io/anlessini/SearchLambda.java]: The response structure is mostly the same, with the only change being the headers. I removed `Access-Control-Allow-Origin` as I find it irrelevant in our scenario, and added `Content-Type=application/json` to indicate the body is JSON text.

Comment (shaneding) [reply src/main/java/io/anlessini/SearchLambda.java]: Ah okay thats good then

**Pull request merged**